### PR TITLE
Smart amp tplg fixes

### PR DESCRIPTION
**Pull request by keyonjie**
make sure the SMART_SSP_QUIRK is configured, and use the unify mclk among SSPs.

This will fix https://github.com/thesofproject/linux/issues/2190

### CODE REVIEW

File: tools/topology/sof-smart-amplifier.m4 [hunk 1]
```diff
@@ -39,6 +39,8 @@ ifdef(`SMART_SSP_INDEX',`',`errprint(note: Need to define SSP index for sof-smar
 # define(`SMART_SSP_NAME', `SSP1-Codec') define SSP BE dai_link name
 ifdef(`SMART_SSP_NAME',`',`errprint(note: Need to define SSP BE dai_link name for sof-smart-amplifier
 )')
+# define(`SMART_SSP_QUIRK', 0) define SSP quirk
```
Comment (lgirdwood) [inline tools/topology/sof-smart-amplifier.m4:42]: Can you comment here "why" - what does the quirk do.? Inline comments will make it easy for users to follow.

File: tools/topology/sof-tgl-max98373-rt5682.m4 [hunk 2]
```diff
@@ -199,7 +201,7 @@ dnl ssp1-maxmspk, ssp0-RTHeadset
 
 #SSP 0 (ID: 0)
 DAI_CONFIG(SSP, 0, 0, SSP0-Codec,
-        SSP_CONFIG(I2S, SSP_CLOCK(mclk, 19200000, codec_mclk_in),
+        SSP_CONFIG(I2S, SSP_CLOCK(mclk, SSP_MCLK, codec_mclk_in),
```
Comment (plbossart) [inline tools/topology/sof-tgl-max98373-rt5682.m4:204]: indeed there are physical restrictions on what can be pushed on the outputs, I don't think anything above 20MHz would work on the MCLK pins.
Can we please check that we never output 38.4 on the MCLK pins in other cases?
Thanks!

File: tools/topology/sof-smart-amplifier.m4 [hunk 3]
```diff
@@ -41,6 +41,8 @@ ifdef(`SMART_SSP_NAME',`',`errprint(note: Need to define SSP BE dai_link name fo
 )')
 # define(`SMART_SSP_QUIRK', 0) define SSP quirk
 ifdef(`SMART_SSP_QUIRK',`',`define(`SMART_SSP_QUIRK', 0)')
+# define(`SSP_MCLK', ) define SSP mclk if not done yet
+ifdef(`SSP_MCLK',`',`define(`SSP_MCLK', 38400000)')
```
Comment (plbossart) [inline tools/topology/sof-smart-amplifier.m4:45]: we don't need to output 38.4 for the codec to work, 19.2 is enough and is safer (will have fewer issues on the line).

File: tools/topology/sof-smart-amplifier.m4 (hunk 1 above)
Comment (keyonjie) [reply tools/topology/sof-smart-amplifier.m4:42]: @lgirdwood it is for the common SSP quirk we have used in other place to configure the SSP specific register bits, e.g. we set LBM(Loop back Mode) for nocodec smart_amp validation in sof-tgl-smart-amp-nocodec.m4.

File: tools/topology/sof-tgl-max98373-rt5682.m4 (hunk 2 above)
Comment (keyonjie) [reply tools/topology/sof-tgl-max98373-rt5682.m4:204]: Thanks for sharing @plbossart 
I am not as surely as you are about that 38.4MHz MCLK doesn't work, we have several places using 38.4, they sounds fine there but maybe the mclk pin could be actually not connected to the codec on those platforms.

File: tools/topology/sof-smart-amplifier.m4 (hunk 3 above)
Comment (keyonjie) [reply tools/topology/sof-smart-amplifier.m4:45]: Good point, let me change the default to 19.2, but keep it configurable from the upper level .m4.

**Comment by keyonjie** [comment]
@lgirdwood @plbossart updated.

**Comment by bardliao** [comment]
Maybe we should use this opportunity to review the mclk value for all existing machines.

**Comment by jajanusz** [comment]
> @lgirdwood @plbossart updated.

@plbossart @lgirdwood please review this PR, we need it for tgl 0.7 release, thanks

**Comment by keyonjie** [comment]
@lgirdwood @plbossart ping for comments, this is fix so in urgent.

**Comment by lgirdwood** [comment]
Jenkins CI know issues.

**Comment by lgirdwood** [comment]
@zrombel looks like CI got stuck, are we good to merge or do you want to rerun ?

**Comment by zrombel** [comment]
Build was re run, and all tests PASS.

**Pull request merged**